### PR TITLE
Clarify docstrings for updated DbApiHook

### PR DESCRIPTION
**Pull request by dstandish**
Started off by fixing typo, replacing `last_statement` with `return_last` but then  tried to clarify the language a bit here and there.

### CODE REVIEW

File: airflow/providers/common/sql/hooks/sql.py [hunk 1]
```diff
@@ -272,33 +273,33 @@ def run(
...
-          * last_description - description of the last statement executed
-
-        Note that return value from the hook will ONLY be actually returned when handler is provided. Setting
-        the ``handler`` to None, results in this method returning None.
+        Note that query result will ONLY be actually returned when a handler is provided; if
+        ``handler`` is None, this method will return None.
 
         Handler is a way to process the rows from cursor (Iterator) into a value that is suitable to be
-        returned to XCom and generally fit in memory. As an optimization, handler is usually not executed
-        by the SQLExecuteQuery operator if `do_xcom_push` is not specified.
```
Comment (dstandish) [inline airflow/providers/common/sql/hooks/sql.py:301]: this bit about sql execute operator, i thought isn't really relevant to hook.run docstring.

**Review by pierrejeambrun — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Line too long :(

**Comment by potiuk** [comment]
Installed pre-commits really save on iterating with CI @dstandish 😛

File: airflow/providers/common/sql/hooks/sql.py [hunk 2]
```diff
@@ -272,33 +273,33 @@ def run(
         where each element in the list are results of one of the queries (typically list of list of rows :D)
 
         For compatibility reasons, the behaviour of the DBAPIHook is somewhat confusing.
-        In cases, when multiple queries are run, the return values will be an iterable (list) of results
-        - one for each query. However, in certain cases, when single query is run - the results will be just
-        the results of that query without wrapping the results in a list.
+        In some cases, when multiple queries are run, the return value will be an iterable (list) of results
+        -- one for each query. However, in other cases, when single query is run, the return value will be just
+        the result of that single query without wrapping the results in a list.
```
Comment (dstandish) [inline airflow/providers/common/sql/hooks/sql.py]:
```suggestion
        In some cases, when multiple queries are run, the return value will be an iterable (list) of results
        -- one for each query. However, in other cases, when single query is run, the return value will
        be the result of that single query without wrapping the results in a list.
```

**Comment by dstandish** [comment]
> Installed pre-commits really save on iterating with CI @dstandish 😛

yeah, sigh... i have them enabled but i have flake8 and mypy disabled because....

iteration time 😎 

just too slow on mac... although.... it was a while ago i last checked

**Comment by potiuk** [comment]
> > Installed pre-commits really save on iterating with CI @dstandish 😛
> 
> yeah, sigh... i have them enabled but i have flake8 and mypy disabled because....
> 
> iteration time 😎
> 
> just too slow on mac... although.... it was a while ago i last checked

Let's speed'em up. On my M1 I have completely no problems with slow iteration time on commits :)

**Comment by potiuk** [comment]
BTW. `SKIP=run-mypy,run-flake8` as global env might help even if there are some slow-downs

**Comment by dstandish** [comment]
> BTW. `SKIP=run-mypy,run-flake8` as global env might help even if there are some slow-downs

yes this is how i have it configured

**Comment by pierrejeambrun** [comment]
Haha I am familiar to this as well 😂.

What I do now is configure mypy and flake as linter in my IDE. This way it is really easy to see/resolve and iterate on typing error. Then the pre-commit hook are just an extra check :)

**Pull request merged**